tools/unstableWarningAnonymizer/unstableAnonScript.chpl

```diff
@@ -31,84 +31,187 @@ features.
+
+  const forallRef ="warning: inferring a 'ref' intent on an array in a forall is unstable - in the future this may require an explicit 'ref' forall intent for";
+  if warning.find(forallRef) != -1 then
+    return forallRef + " <var-name>";
+
+  const typeName = "warning: using a type's name ";
+  const typeNameUse = "in a 'use' statement to access its tertiary methods is an unstable feature";
+  const typeNameImport = "in an 'import' statement to access its tertiary methods is an unstable feature";
+  if warning.find(typeName) != -1 && warning.find(typeNameUse) != -1 then
+    return typeName + typeNameUse;
+  if warning.find(typeName) != -1 && warning.find(typeNameImport) != -1 then
+    return typeName + typeNameImport;
+
+  const underscore = "warning: symbol names with leading underscores";
+  const end = " are unstable";
+  if warning.find(underscore) != -1 && warning.find(end) != -1 then
+    return underscore + end;
+
+  const chpl_ = "warning: symbol names beginning with 'chpl_' ";
+  if warning.find(chpl_) != -1 && warning.find(end) != -1 then
+    return chpl_ + end;
+
+  const pragmas = "uses pragmas, which are considered unstable and may change in the future";
+  if warning.find(pragmas) != -1 then
+    return "warning: <proc> " + pragmas;
+
+  return warning;
+}
+
+inline proc prettyPrintArr(arr : [] (string, int, int), writer: IO.fileWriter(?), fileCount: bool){
+  for a in arr {
+    const grammar = if a[1] < 2 then " instance of \"" else " instances of \"";
+    const files = if fileCount then
+                  "\" across " + a[2] :string + if a[2] < 2 then
+                  " file" else " files"
+                  else "\"";
+    writer.writeln(a[1], grammar, a[0], files);
+  }
+}
+
+
+inline proc csvPrintArr(arr : [] (string, int, int), writer: IO.fileWriter(?), fileCount: bool){
+  writer.writeln("warning", ",", "count", if fileCount then ",uniqueFiles" else "");
+  for a in arr {
+    const files = if fileCount then
+                  "," + a[2]:string
+                  else "";
+    writer.writeln("\"", a[0], "\"", ",", a[1], files);
   }
 }
 
+// Comparator to sort our array representation of the map
+// by the number of occurences of each warning
+record OccurenceComparator {}
+// a and b will be (string, int, int)
+proc OccurenceComparator.compare(a, b) { return b[1] - a[1] ; } // Reverse sort
 
-inline proc csvPrintMap(m : map(?), writer: IO.fileWriter(?)){
+record WarningComparator {}
+proc WarningComparator.key(a) { return a[0]; }
+
+proc convertMapToArray(const m: map(string, ?t), sorted: bool, topX: int) where t == (int, set(string)){
+  var arr : [0..<m.size] (string, int, int);
+  var i = 0;
   for key in m.keys() {
-    writer.writeln("\"", key, "\"", ",", m[key]);
+    arr[i] = (key, m[key][0], m[key][1].size); // We don't need to save the entire list
+                                        // of fileNames at this point, just the numer is enough
```

Comment:
Typo: "numer" should be "number"?

Also, you could just put the whole comment on the line before instead of having to split it up (but in the grand scheme of things, this is not super important)